src/lib/structures/Command.ts

```diff
@@ -72,10 +74,18 @@ export abstract class Command extends BaseAliasPiece {
 		this.guarded = options.guarded!;
 		this.hidden = options.hidden!;
 		this.flags = options.flags!;
-		this.quotedStringSupport = options.quotedStringSupport!;
+		this.quotes = options.quotes ?? [];
 	}
 
-	public abstract run(message: Message): Awaited<unknown>;
+	public preParse(message: Message, commandName: string, prefix: string): Awaited<unknown> {
+		const input = message.content.substr(prefix.length + commandName.length);
+		const lexer = new Lexure.Lexer().setInput(input).setQuotes(this.quotes);
```

Comment:
wouldn't it be better to do the `new Lexure.Lexer()` part as a private property so it doesn't instantiate it every time this function is called?

So

```ts

#lexer = new Lexure.Lexer()

// and here
const lexer = this.#lexer.setInput(input).setQuotes(this.quotes)
```